integration-tests/tests/src/tests/queries.ts

```diff
@@ -195,14 +198,29 @@ type QueryResultsPair = [ExpectedResults: any[], Query: string, ...QueryArgs: Ar
  * Helper method which runs an array of queries and asserts them to given expected length
  * For example: (r, "Obj", [1, "intCol == 0"]) => querying "intCol == 0" should return 1 elements.
  */
-const expectQueryLength = (realm: Realm, objectSchema: Realm.ObjectClass, queryLengthPairs: QueryLengthPair[]) => {
-  queryLengthPairs.forEach(([expectedLength, queryString, ...queryArgs]) => {
-    const filtered = realm.objects(objectSchema).filtered(queryString, ...queryArgs);
-    expect(filtered.length).equal(
-      expectedLength,
-      `Expected length ${expectedLength} for query: ${queryString} ${JSON.stringify(queryArgs)}`,
-    );
-  });
+const expectQueryLength = async (
+  realm: Realm,
+  objectSchema: Realm.ObjectClass,
+  queryLengthPairs: QueryLengthPair[],
+  useFlexibleSync = false,
+) => {
+  return Promise.all(
+    queryLengthPairs.map(async ([expectedLength, queryString, ...queryArgs]) => {
```

Comment:
Fun fact: The switch from `forEach()` to `map()` shouldn't be a mocha thing as the JS `forEach()` doesn't wait for promises when calling its callback.